front/components/DataSourceViewResourceSelectorTree.tsx

```diff
@@ -216,6 +228,38 @@ function DataSourceViewResourceSelectorChildren({
             />
           );
         })}
+        {dataSourceView.category === "managed" && nodes.length === 0 && (
+          <div className="flex w-full flex-col items-center gap-2 rounded-lg border bg-structure-50 py-2">
+            <span className="text-element-700">The Vault is empty!</span>
+            {owner.role === "admin" && connector ? (
+              <>
+                <Button
+                  label="Add Data"
+                  icon={PlusIcon}
+                  onClick={() => {
+                    setShowConnectorPermissionsModal(true);
+                  }}
+                />
+                <ConnectorPermissionsModal
```

Comment:
I'm definitely not a big fan of embedding this within the tree. The goal of this tree was to be easy to use in many different places (poke, simply viewing available nodes, ...). This component is pretty heavy and load a shit ton of other components, any chance we can do something a bit more light so we can use this component in more places?

Reply:
@JulesBelveze what about an optional arg for a callback if there is no content ?
If the arg is present, show a button that callback() otherwise, none.

Reply:
That way you can leave everything else outside of this component.

Reply:
We can definitely do that, but if we are willing to instantiate that modal in a place that makes we will have to pass that callback around a bunch of components..

Reply:
Fair point :)
Maybe create a component around this one with the modal, hooks etc?

Reply:
Also, this button needs to be there in every instance of the selector tree as we want people to be able to request/add data if the managed connection doesn't have data in it. I feel like this is a lot of energy for a very edge case..

Reply:
> Maybe create a component around this one with the modal, hooks etc?

Will do this and fix @flvndvd comment on the `useConnector` as a first iteration 😄